### PR TITLE
build: Added module for generating docs from notebooks

### PR DESCRIPTION
**Pull request by BrendanWalsh**
Replace the existing convertNotebooks.py with an extensible python module for creating notebook processing pipelines, speeding up website document generation by ~20x

The new documentprojection module allows for the creation of pipelines where notebooks are tested, reformatted to arbitrary formats and published to arbitrary targets such as a custom website, Azure Docs, etc.

```
(synapseml) <user>:~/dev/SynapseML/docs/python$ python documentprojection --help
usage: documentprojection [-h] [-f] [-p] [-m] [-r] [-l] [-c {all,website,console,azure_docs}] [-v {debug,info,warn,error,critical}] [N [N ...]]

Document Projection Pipeline

positional arguments:
  N                     a notebook or folder of notebooks to project

optional arguments:
  -h, --help            show this help message and exit
  -f, --format          run only formatters
  -p, --publish         run publishers. forces -t and -f.
  -m, --metadata        format documents and only output metadata of formated documents.
  -r, --recursive       recursively scan for notebooks in the given directory.
  -l, --list            list notebooks
  -c {all,website,console,azure_docs}, --channel {all,website,console,azure_docs}
                        the channel through which the notebook(s) should be processed. defaults to all if not specified.
  -v {debug,info,warn,error,critical}, --loglevel {debug,info,warn,error,critical}
                        set log level
```

### CODE REVIEW

File: docs/python/documentprojection/__main__.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    parser.add_argument(
+        "notebooks",
+        metavar="N",
+        type=str,
+        nargs="*",
+        help="a notebook or folder of notebooks to project",
+    )
+    parser.add_argument(
+        "-t", "--test", action="store_true", default=False, help="run only tests"
+    )
```
Comment (mhamilton723) [inline docs/python/documentprojection/__main__.py:42]: Does this system need to run tests? Right now thats handled by scala so you might not need to worry about this

**Comment by BrendanWalsh** [comment]
Removed incomplete code planned for future commits.

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**